server/src/test/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitorTests.java

```diff
@@ -722,6 +718,111 @@ protected void setIndexCreateBlock(ActionListener<Void> listener, boolean indexC
         assertTrue(countBlocksCalled.get() == 0);
     }
 
+    public void testIndexCreateBlockRemovedOnlyWhenAnyNodeAboveHighWatermark() {
+        AllocationService allocation = createAllocationService(
+            Settings.builder()
+                .put("cluster.routing.allocation.node_concurrent_recoveries", 10)
+                .put("cluster.blocks.create_index.enabled", false)
+                .build()
+        );
+        Metadata metadata = Metadata.builder()
+            .put(
```

Comment:
would like some comments explaining what these 3 tests are doing

Reply:
Added comments ([Background Context](https://github.com/opensearch-project/OpenSearch/pull/5852)). Essentially the test validates that IndexCreateBlock only gets removed when any of the node is no longer breaching high watermark. Essentially earlier issue was this block was removed even though all the nodes were breaching high watermark, as condition for block removal was IndexCreateBlock should be present on the node and auto release of this block should be enabled (which is true by default). So if an IndexCreateBlock got added on the cluster (due to all node breaching high watermark), in the subsequent evaluation of DiskThresholdMonitor (after 15 sec), the earlier unblock condition will be true (even though all nodes are breaching high watermark) and IndexCreateBlock will be removed.

Let me know if this clarifies the isssue.